src/main/java/Duke.java

```diff
@@ -1,10 +1,265 @@
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * The main class of the project.
+ */
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    private Storage storage;
+    private TaskList taskList;
+
+    /**
+     * Constructs a new instance of Duke.
+     *
+     * @param taskList the task list
+     */
+    public Duke(TaskList taskList) {
+
+        storage = new Storage(Paths.get(System.getProperty("user.home"), "data", "duke.txt"));
+        this.taskList = taskList;
+        storage.loadTasks(taskList);
     }
+
+    /**
+     * Gets Duke's task list.
+     *
+     * @return Duke's task list.
+     */
+    public TaskList getTaskList() {
+
+        return taskList;
+    }
+
+    /**
+     * Executes the command.
+     *
+     * @param command the string command supplied by the user.
+     * @return the relevant response after processing the command.
+     */
+    public Response executeCommand(String command) {
```

Comment:
This method is really lengthy which is makes it very hard to understand. Think of possible ways to reduce the length. e.g. take out repeated code segment or specific functional code as methods.